src/detect-engine.c

```diff
@@ -3276,6 +3276,10 @@ TmEcode DetectEngineThreadCtxInit(ThreadVars *tv, void *initdata, void **data)
     det_ctx->counter_alerts = StatsRegisterCounter("detect.alert", tv);
     det_ctx->counter_alerts_overflow = StatsRegisterCounter("detect.alert_queue_overflow", tv);
     det_ctx->counter_alerts_suppressed = StatsRegisterCounter("detect.alerts_suppressed", tv);
+
+    /* Register counter for Lua rule errors. */
+    det_ctx->lua_rule_errors = StatsRegisterCounter("detect.lua_errors", tv);
```

Comment:
would it make sense to have a lua name space? `detect.lua.errors`? Could imagine having more, like `detect.lua.evals` or `detect.lua.matches`, etc

Reply:
Yes, does make sense. We'll also have errors due to resource/instruction limits as well. Will be nice to separate those out.